src/charge_controller/dev/ControllerUI.cpp

```diff
@@ -0,0 +1,106 @@
+#include <charge_controller/dev/ControllerUI.hpp>
+
+ControllerUI::ControllerUI(DEV::Encoder& encoder, DEV::Button& encoderButton, LCDDisplay& display, ControllerModel& model) : model(model), encoder(encoder), encoderButton(encoderButton), display(display) {
+    encoder.setRangeAndPosition(model.getEncoderRange(ControllerModel::State::PAGE_SELECT), 0);
+}
+
+void ControllerUI::process() {
```

Comment:
This function isn't too bad, but it's getting clos to the point that it would be worth making helper methods for each state. You can do this now or just leave it alone

Reply:
added some helper methods to break it out.